internal/api/auth.go

```diff
@@ -0,0 +1,131 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"time"
+
+	"golang.org/x/crypto/bcrypt"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+type Creds struct {
+	User string `json:"login"`
+	Pwd  string `json:"password"`
+}
+
+type commonAuth func(creds Creds) (userID int, httpCode int, err error)
+
+func (h *Handler) Register(w http.ResponseWriter, r *http.Request) {
+	AuthCmnHandler(w, r,
+		func(creds Creds) (userID int, httpCode int, err error) {
+			userID, err = newUser(h.store, creds)
+			httpCode = http.StatusConflict
+			return userID, httpCode, err
+		},
+	)
+}
+
+func (h *Handler) Login(w http.ResponseWriter, r *http.Request) {
+	AuthCmnHandler(w, r,
+		func(creds Creds) (userID int, httpCode int, err error) {
+			userID, err = authUser(h.store, creds)
+			httpCode = http.StatusUnauthorized
+			return userID, httpCode, err
+		},
+	)
+}
+
+func AuthCmnHandler(w http.ResponseWriter, r *http.Request, auth commonAuth) {
+	if r.Header.Get("Content-Type") != "application/json" ||
+		r.Method != http.MethodPost {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	creds := Creds{}
+	var buf bytes.Buffer
+	// читаем тело запроса
+	_, err := buf.ReadFrom(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	if err = json.Unmarshal(buf.Bytes(), &creds); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	if creds.User == "" || creds.Pwd == "" {
+		http.Error(w, "empty login or password", http.StatusBadRequest)
+		return
+	}
+	var userID, httpErrCode int
+	userID, httpErrCode, err = auth(creds)
+	if err != nil {
+		http.Error(w, err.Error(), httpErrCode)
+		return
+	}
+
+	var tkn string
+	tkn, err = BuildJWT(userID)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Authorization", "Bearer "+tkn)
```

Comment:
 Можно подумать о том чтобы вынести эту часть так же в middleware.

```suggestion
	var tkn string
	tkn, err = BuildJWT(userID)
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
	w.Header().Set("Authorization", "Bearer "+tkn)
```